src/main/java/seedu/duke/parser/commandparser/UpdateParser.java

```diff
@@ -1,11 +1,215 @@
+        } catch (IndexOutOfBoundsException e) {
+            Ui.printLineBreak();
+            System.out.println("That selection does not exist!");
+        }
         return null;
     }
-}
+
+    private static Command parseUpdateEvent(String[] userUpdates, Event event)
+            throws DukeException, InvalidBudgetException {
+        String[] parsedAttributes = new String[3];
+        LocalDateTime dateTime = null;
+        double budget = 0;
+        boolean isFlag;
+        for (String update : userUpdates) {
+            isFlag = false;
+            String[] attribute = update.trim().split("/+");
+            if (update.contains(TITLE_FLAG)) {
+                isFlag = true;
+                parsedAttributes[INDEX_OF_TITLE] = attribute[1];
+            }
+            if (update.contains(DATE_FLAG)) {
+                isFlag = true;
+                dateTime = Parser.convertDateTime(attribute[1]);
+            }
+            if (update.contains(VENUE_FLAG)) {
+                isFlag = true;
+                parsedAttributes[INDEX_OF_VENUE] = attribute[1];
+            }
+            if (update.contains(BUDGET_FLAG)) {
+                isFlag = true;
+                budget = convertEventBudgetToDouble(attribute[1]);
+            }
+            if (update.contains(DESCRIPTION_FLAG)) {
+                isFlag = true;
+                parsedAttributes[INDEX_OF_DESCRIPTION] = attribute[1];
+            }
+            if (update.contains(TASK_FLAG)) {
+                attribute[1] = attribute[1].replaceAll("\\s", "");
+                return parseUpdateTask(event, attribute[1]);
+            }
+            if (!isFlag) {
+                System.out.println("Invalid update, you have returned to the main page!");
+            }
+        }
+        return new UpdateEventCommand(event, parsedAttributes, dateTime, budget);
+    }
+
+    protected static Command parseUpdateTask(Event eventToBeUpdated, String index)
+            throws DukeException {
+        int taskNum = Integer.parseInt(index) - 1;
+        Task taskToBeUpdated = eventToBeUpdated.getFromTaskList(taskNum);
+        String[] userUpdates = prepareTaskUpdates(taskToBeUpdated);
+
+        String[] parsedAttributes = new String[3];
+        LocalDateTime dateTime = null;
+
+        for (String update : userUpdates) {
+            String[] attribute = update.trim().split("/+");
+            if (update.contains(TITLE_FLAG)) {
+                parsedAttributes[INDEX_OF_TITLE] = attribute[1];
+            } else if (update.contains(DEADLINE_FLAG)) {
+                dateTime = Parser.convertDateTime(attribute[1]);
+            } else if (update.contains(DESCRIPTION_FLAG)) {
+                parsedAttributes[INDEX_OF_DESCRIPTION] = attribute[1];
+            } else if (update.contains(MEMBER_FLAG)) {
+                attribute[1] = attribute[1].replaceAll("\\s", "");
+                return changeMember(attribute[1], taskToBeUpdated);
```

Comment:
It looks like this regex phrase replaces all whitespace/consecutive whitespaces from the attribute (containing the index of the member) with an empty string. Is there any difference between this and performing a `trim()` operation on the string?

Reply:
performing a trim(), does not remove the whitespace within a given cell in an index, hence to remove the whitespace, I am required to perform a replace all